### ISSUE TITLE
java.lang.NoClassDefFoundError: antlr/collections/AST in Eclipse plugin

### ISSUE THREAD

**Issue by PhilippWendler** [sevntu-checkstyle/sevntu.checkstyle#460]
I am using Eclipse Neon 4.6.0 with the EclipseCS plugin 6.19.0 and the SevNTU-Checks plugin 1.21.0.

For some classes in my workspace, I get the following error when the project is build (shown in Eclipse error log):

```
java.lang.Error: Error was thrown while processing [.../MyClass.java]
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:285)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:196)
	at net.sf.eclipsecs.core.builder.Auditor.runAudit(Auditor.java:159)
	at net.sf.eclipsecs.core.builder.CheckstyleBuilder.handleBuildSelection(CheckstyleBuilder.java:300)
	at net.sf.eclipsecs.core.builder.CheckstyleBuilder.build(CheckstyleBuilder.java:171)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:360)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:383)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:144)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:235)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
Caused by: java.lang.NoClassDefFoundError: antlr/collections/AST
	at com.github.sevntu.checkstyle.checks.design.PublicReferenceToPrivateTypeCheck.getMethodOrFieldReferencedTypes(PublicReferenceToPrivateTypeCheck.java:208)
	at com.github.sevntu.checkstyle.checks.design.PublicReferenceToPrivateTypeCheck.addExternallyAccessibleMethodTypes(PublicReferenceToPrivateTypeCheck.java:175)
	at com.github.sevntu.checkstyle.checks.design.PublicReferenceToPrivateTypeCheck.visitToken(PublicReferenceToPrivateTypeCheck.java:120)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:362)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:505)
	at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:304)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:179)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:78)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:301)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:271)
	... 16 more
Caused by: java.lang.ClassNotFoundException: antlr.collections.AST cannot be found by eclipsecs-sevntu-plugin_1.21.0
	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:448)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:361)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:353)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:161)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 26 more
```

I am not sure how I could make this reproducible for others. What would you need for this?

**Comment by vorburger** [comment]
@PhilippWendler and FYI @romani this one is most likely my fault... :-( Sorry! Please see https://sourceforge.net/p/eclipse-cs/bugs/403/ and https://sourceforge.net/p/eclipse-cs/bugs/407/ for background. I believe there is nothing particular TBD in sevntu.checkstyle about this; instead watch eclipse-cs/bugs/407, I'll try to propose a better solution there, and then this would hopefully eventually get fixed upstream in a new version of eclipse-cs (perhaps depending on a new version of core Checkstyle incl. https://github.com/checkstyle/checkstyle/issues/3169). 

Up to you if you want to keep this bug open just as a reminder and for others to find it, or close it as this isn't actually specific to sevntu.checkstyle. (All Checkstyle Extension Eclipse plugins are broken with EclipseCS plugin 6.18.0 & 6.19.0 because of https://sourceforge.net/p/eclipse-cs/bugs/403/; with 6.17.0 and earlier you should not have this problem.)

**Comment by PhilippWendler** [comment]
Fixed with EclipseCS plugin 6.19.1 (https://sourceforge.net/p/eclipse-cs/bugs/407/#c5f4).
